src/main/java/com/strandls/cca/controller/CCADataController.java

```diff
@@ -485,4 +487,28 @@ public Response searchmapCCAData(@QueryParam("query") String query, @Context Htt
 		}
 	}
 
+
+
+@POST
+@Path(ApiConstants.BULK_UPLOAD)
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@ValidateUser
+@ApiOperation(value = "Save multiple CCA data", notes = "Saves multiple CCA data entries", response = CCAData.class, responseContainer = "List")
+@ApiResponses(value = { @ApiResponse(code = 404, message = "Could not save the data", response = String.class) })
+
+public Response saveBulkCCAData(@Context HttpServletRequest request, List<CCAData> ccaDataList)
+        throws CCAException {
+    try {
+        List<CCAData> savedDataList = new ArrayList<>();
+        for (CCAData ccaData : ccaDataList) {
```

Comment:
1. Write a separate function (something like `saveCCADataInBulk`) inside `CCADataServiceImpl.java` class which is handles the business logic of the code.
2. Call that function from the controller class and return the result.

**Reason**:We should try to write the business logic inside functions defined for the service interface implementation classes. The controller is mostly responsible for directing the request to the relevant functions.